.github/workflows/pr-realm-js.yml

```diff
@@ -189,68 +214,172 @@ jobs:
           echo 'UseMultiToolTask=true' >> $GITHUB_ENV
           echo 'VCPKG_KEEP_ENV_VARS=CCACHE_COMPILER;CCACHE_STATSLOG;CCACHE_LOGFILE;CCACHE_COMPILERTYPE;UseMultiToolTask' >> $GITHUB_ENV
 
-      # build the c++ library for standard targets
-      - name: Build node
-        if: ${{ matrix.variant.os != 'ios' && matrix.variant.os != 'android' }}
-        run: npm run build:node:prebuild:${{matrix.variant.arch}} --workspace realm
-
-      # build the c++ library for IOS
-      # the Info.plist needs to be regenerated with all libraries in place
-      - name: Build iOS
-        if: ${{ matrix.variant.os == 'ios' }}
-        run: |
-          npm run build:ios --workspace realm
-          rm -vf ${{ matrix.variant.artifact-path }}/Info.plist
+      - name: Build native module
+        run: npm run prebuild-node --workspace realm
         env:
-          PLATFORMS: ${{ matrix.variant.arch }}
-
-      # build the c++ library for Android
-      - name: Build Android
-        if: ${{ matrix.variant.os == 'android' }}
-        run: npm run build:android --workspace realm -- --arch=${{matrix.variant.arch}}
+          PREBUILD_ARCH: ${{ matrix.arch }}
 
-      # Due to a limitation in upload-artifact a redundant file is needed to force
-      # preserving paths (https://github.com/actions/upload-artifact/issues/174)
       - name: Upload prebuild artifact
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
```

Comment:
Closes https://github.com/realm/realm-js/issues/6530

Reply:
Not entirely, but it does bring us closer. Before closing that issue, I'd like for the test jobs to be split out and only depend on the artifacts that they actually need:
> refactor our workflow into jobs where a test run only depends on the upload of the prebuilt artefact that it actually needs to complete.